File: frontend/src/types/application.ts
/**
 * TypeScript interfaces mirroring backend/app/models/application.py
 */

export enum GenerationStatus {
  PENDING = 'pending',
  MATCHING = 'matching',
  GENERATING_POINTS = 'generating_points',
  WRITING_RESUME = 'writing_resume',
  RENDERING_LATEX = 'rendering_latex',
  COMPLETED = 'completed',
  FAILED = 'failed',
}

export interface BulletPoint {
  id: string;
  section: string;
  text: string;
  order: number;
  edited: boolean;
}

export interface SectionPoints {
  section_key: string;
  section_title: string;
  bullets: BulletPoint[];
}

export interface GeneratedContent {
  resume_points: SectionPoints[];
  resume_latex?: string;
  model_used: string;
}

export interface Application {
  id: string;
  created_at: string;
  updated_at: string;
  company_name: string;
  company_description?: string;
  job_title: string;
  job_description: string;
  selected_project_ids: string[];
  generation_status: GenerationStatus;
  generated?: GeneratedContent;
  error_message?: string;
}

File: backend/app/models/application.py
"""
Application / generation history models.
"""
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from enum import Enum


class GenerationStatus(str, Enum):
    PENDING = "pending"
    MATCHING = "matching"
    GENERATING_POINTS = "generating_points"
    WRITING_RESUME = "writing_resume"
    RENDERING_LATEX = "rendering_latex"
    COMPLETED = "completed"
    FAILED = "failed"


class BulletPoint(BaseModel):
    id: str
    section: str
    text: str
    order: int = 0
    edited: bool = False


class SectionPoints(BaseModel):
    section_key: str
    section_title: str
    bullets: List[BulletPoint] = []


class GeneratedContent(BaseModel):
    resume_points: List[SectionPoints] = []
    resume_latex: Optional[str] = None
    model_used: str = ""


class Application(BaseModel):
    id: str
    created_at: datetime
    updated_at: datetime
    company_name: str
    company_description: Optional[str] = None
    job_title: str
    job_description: str
    selected_project_ids: List[str] = []
    generation_status: GenerationStatus = GenerationStatus.PENDING
    generated: Optional[GeneratedContent] = None
    error_message: Optional[str] = None
